### PR TITLE
KAFKA-6753: Update controller metrics periodically instead of after processing every event

### PR DESCRIPTION
**Pull request by gitlw**
Update controller metrics periodically, e.g. once per second, instead of after processing every event

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -344,6 +345,7 @@ object KafkaConfig {
   val AutoLeaderRebalanceEnableProp = "auto.leader.rebalance.enable"
   val LeaderImbalancePerBrokerPercentageProp = "leader.imbalance.per.broker.percentage"
   val LeaderImbalanceCheckIntervalSecondsProp = "leader.imbalance.check.interval.seconds"
+  val ControllerUpdateMetricsIntervalMsProp = "controller.update.metrics.interval.ms"
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:348]: Adding a config requires a KIP. Is there really a need for a config?

**Review by ijuma — COMMENTED** [review]
Nice find! Do any tests need to be updated?

**Comment by ijuma** [comment]
One more thing: one of the metrics is the event we are currently processing. It seems a bit odd to only update that once a second. Have we done an analysis on the existing metrics to check if this makes sense for all of them?

**Comment by gitlw** [comment]
@ijuma I discussed with @becketqin about this PR, and we agree that an alternative, and also probably better, way of solving the problem is to update the metrics only when necessary instead of letting the controller blindly update the metrics. So I'll close this PR and create another one using that approach.

**Pull request closed without merging**